modules/light-clients/08-wasm/types/upgrade.go

```diff
@@ -36,8 +36,8 @@ func (cs ClientState) VerifyUpgradeAndUpdateState(
 
 	payload := SudoMsg{
 		VerifyUpgradeAndUpdateState: &VerifyUpgradeAndUpdateStateMsg{
-			UpgradeClientState:         *wasmUpgradeClientState,
-			UpgradeConsensusState:      *wasmUpgradeConsState,
+			UpgradeClientState:         wasmUpgradeClientState.Data,
+			UpgradeConsensusState:      wasmUpgradeConsState.Data,
```

Comment:
I think we must pass the entire client state here not ` wasmUpgradeClientState.Data`. Because it carries the latest height. And this is a must have information.

Reply:
This doesn't make sense to me. The new client state must overwrite the old one. Including the latest height right? [We do this in tendermint client](https://github.com/cosmos/ibc-go/blob/c54ec070d0c4fb7e6b923d2c1d475fd2e6011254/modules/light-clients/07-tendermint/upgrade.go#L100).

And implicitly, isn't it used [here](https://github.com/misko9/centauri/blob/4a9a3313e1f925ebef5e6c4590b87cbf621679fc/light-clients/ics10-grandpa/src/client_def.rs#L419) in the grandpa client?

Reply:
> And implicitly, isn't it used [here](https://github.com/misko9/centauri/blob/4a9a3313e1f925ebef5e6c4590b87cbf621679fc/light-clients/ics10-grandpa/src/client_def.rs#L419) in the grandpa client?

The `upgraded_client_state` that is used in this line is actually the underlying client state (so not the 08-wasm wrapping client state). You can see that it is retrieved [here](https://github.com/misko9/centauri/blob/4a9a3313e1f925ebef5e6c4590b87cbf621679fc/light-clients/ics10-grandpa-cw/src/contract.rs#L316) (the [`try_from` function will unwrap the underlying client state and return it](https://github.com/misko9/centauri/blob/4a9a3313e1f925ebef5e6c4590b87cbf621679fc/light-clients/ics10-grandpa-cw/src/msg.rs#L375)).

Reply:
I see. But is this what they should be doing? Then why is the 07-tendermint client written this way?

Reply:
Contracts should be writing a 08-wasm client with the updated latest height, but the latest height can absolutely be obtained from `wasmUpgradeClientState.Data` (it will be in the underlying client state)

Reply:
Ah, this makes sense. It's good then